tests/network/vmi_subdomain.go

```diff
@@ -80,18 +83,62 @@ var _ = SIGDescribe("Subdomain", func() {
 			Expect(err).ToNot(HaveOccurred())
 			vmi = tests.WaitUntilVMIReady(vmi, console.LoginToFedora)
 
-			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
-				&expect.BSnd{S: "\n"},
-				&expect.BExp{R: console.PromptExpression},
-				&expect.BSnd{S: "hostname -f\n"},
-				&expect.BExp{R: expectedFQDN},
-			}, 10)).To(Succeed(), "failed to get expected FQDN")
+			Expect(assertFQDNinGuest(vmi, expectedFQDN)).To(Succeed(), "failed to get expected FQDN")
 		},
 			table.Entry("with Masquerade binding and subdomain", fedoraMasqueradeVMI, subdomain),
 			table.Entry("with Bridge binding and subdomain", fedoraBridgeBindingVMI, subdomain),
 			table.Entry("with Masquerade binding without subdomain", fedoraMasqueradeVMI, ""),
 			table.Entry("with Bridge binding without subdomain", fedoraBridgeBindingVMI, ""),
 		)
+
+		It("VMI with custom DNSPolicy should have the expected FQDN", func() {
+			vmiSpec := fedoraBridgeBindingVMI()
+			vmiSpec.Spec.Subdomain = subdomain
+			expectedFQDN := fmt.Sprintf("%s.%s.%s.svc.cluster.local", vmiSpec.Name, subdomain, util.NamespaceTestDefault)
+			vmiSpec.Labels = map[string]string{selectorLabelKey: selectorLabelValue}
+
+			dnsServerIP, err := libnet.ClusterDNSServiceIP()
+			Expect(err).ToNot(HaveOccurred())
+
+			vmiSpec.Spec.DNSPolicy = "None"
+			vmiSpec.Spec.DNSConfig = &k8sv1.PodDNSConfig{
+				Nameservers: []string{dnsServerIP},
+				Searches: []string{util.NamespaceTestDefault + ".svc.cluster.local",
+					"svc.cluster.local", "cluster.local", util.NamespaceTestDefault + ".this.is.just.a.very.long.dummy"},
+			}
+
+			vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmiSpec)
+			Expect(err).ToNot(HaveOccurred())
+			vmi = tests.WaitUntilVMIReady(vmi, console.LoginToFedora)
+
+			Expect(assertFQDNinGuest(vmi, expectedFQDN)).To(Succeed(), "failed to get expected FQDN")
+		})
+	})
+
+	It("VMI with custom DNSPolicy and a subdomain should have the expected resolv.conf", func() {
```

Comment:
It took me some time to understand that this test is having a subdomain configured on the VMI but as there is no headless service, it is not expected to be added to the guest. Please confirm that I understand correctly.
Please try to make this clearer in the text.

The second point I have, is about the differences from the other test you added.
This one uses `resolv.conf` while the other`hostname` to check for the FQDN. What is the reason for the difference?
Also, consider extracting the assertion here to a function like the other one.

Reply:
> It took me some time to understand that this test is having a subdomain configured on the VMI but as there is no headless service, it is not expected to be added to the guest. Please confirm that I understand correctly.

Exactly, as we can't update any entry with subdomain, just specific ones.

> The second point I have, is about the differences from the other test you added. This one uses `resolv.conf` while the other`hostname` to check for the FQDN. What is the reason for the difference? 

You can't check FQDN when you don't have the needed entries (`subdomain.namespace.svc.cluster.local`), it will be just empty or equal VM name.
So this tests checks the exact thing it should which is that the guest `/etc/resolv.conf` has the expected `search` line

Reply:
Thank you for the clarification.

I think it will be beneficial if the test name/text expresses what is expected here given the state that affects it most.
Requiring one to delve into the test code to understand basic expectations is not nice.

What do you think about this: `VMI with custom DNSPolicy, a subdomain and no service entry, should not include the subdomain in the searchlist`.

Regarding testing the `hostname` and `resolv.conf`: Sorry, I still do not understand why you could not use the same. On one you do get what you expect, on the other you do not (expect not to get it). You could also perform both checks if you think it is better. My main point is that having two ways to check something is confusing and not consistent, it will be better to show we are checking the same with different expectations.

Reply:
Adding the expect not to get it, 
Note that there is also no headless service there, so it is not the only reason we won't get it